### DISCUSSION TITLE
Manual sidecar gives grpc error

### DISCUSSION THREAD

**Discussion by maraiskruger1980** [jaegertracing/jaeger#3064] [Q&A]
I configured a  manual sidecar:

  name: jaeger-agent
  image: jaegertracing/jaeger-agent:1.23.0
  args:
    - --reporter.type=grpc
    - --reporter.grpc.host-port=dns:///jaeger-collector-headless.monitoring:14250
   

I have the jaeger collector 1.23.0  running in the "monitoring" namespace.

I used Helm and not the operator.

Any ideas what I did wrong?

On the agent sidecar I get the following repeating error: 
`{"level":"error","ts":1622995683.115122,"caller":"grpc/reporter.go:74","msg":"Could not send spans over gRPC","error":"rpc error: code = Unavailable desc = last resolver error: produced zero addresses","stacktrace":"github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).send\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:74\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:53\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*MetricsReporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/metrics.go:85\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*ClientMetricsReporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/client_metrics.go:121\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*agentProcessorEmitBatch).Process\n\tgithub.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:176\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*AgentProcessor).Process\n\tgithub.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:123\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.(*ThriftProcessor).processBuffer\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:122\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.NewThriftProcessor.func2\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:87"}
`

**Comment by maraiskruger1980** [comment]
> I configured a manual sidecar:
> 
> name: jaeger-agent
> image: jaegertracing/jaeger-agent:1.23.0
> args:
> - --reporter.type=grpc
> - --reporter.grpc.host-port=dns:///jaeger-collector-headless.monitoring:14250
> 
> I have the jaeger collector 1.23.0 running in the "monitoring" namespace.
> 
> I used Helm and not the operator.
> 
> Any ideas what I did wrong?
> 
> On the agent sidecar I get the following repeating error:
> `{"level":"error","ts":1622995683.115122,"caller":"grpc/reporter.go:74","msg":"Could not send spans over gRPC","error":"rpc error: code = Unavailable desc = last resolver error: produced zero addresses","stacktrace":"github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).send\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:74\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:53\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*MetricsReporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/metrics.go:85\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*ClientMetricsReporter).EmitBatch\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/client_metrics.go:121\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*agentProcessorEmitBatch).Process\n\tgithub.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:176\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*AgentProcessor).Process\n\tgithub.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:123\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.(*ThriftProcessor).processBuffer\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:122\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.NewThriftProcessor.func2\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:87"} `

Update: If I put the ip address of the collector, it works. If I put the dns addess, then it doesn't work
e.g. 120.64.98.189 14250 works
jaeger-collector.observibility.svc.cluster.local:14250 doesn't work

**Reply by jpkrohling** [reply]
Can you ping the collector hostname from a container in your namespace? If you can't, the problem is not on Jaeger's nor the operator's side.

**Reply by maraiskruger1980** [reply]
it was my mistake. Had a whitespace in front of the address
